pkg/hubble/filters/patterns.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filters
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	// fqdnRegexpStr matches an FQDN, inluding underscores.
+	// FIXME this should not match components that begin or end with hyphens, e.g. -foo-
+	fqdnRegexpStr = `(?:[-0-9_a-z]+(?:\.[-0-9_a-z]+)*)`
```

Comment:
nit: How about this then:

```suggestion
	fqdnRegexpStr = `(?:[0-9_a-z][-0-9_a-z]*(?:\.[-0-9_a-z]+)*)`
```

Reply:
This suggestion still allows the second and subsequent domain name components to start and end with a `-`, and for the first component to end with a `-`, so I don't think it's a significant improvement.